build/commands/lib/genGradle.js

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+const config = require('./config')
+const util = require('./util')
+const Log = require('./logging')
+
+let options = config.defaultOptions
+options.continueOnFail = false
+
+const genGradle = (passthroughArgs, build_dir) => {
+  Log.progressScope('Generating Gradle files', () => {
+    args = [
+      'build/android/gradle/generate_gradle.py',
+      '--output-directory',
+      build_dir
```

Comment:
the output directory can be derived the same way we generate `android_Component_arm64` and other similar dirs.

so it could be nice if a command looks like `build, test` commands:
`npm run gen_gradle Component -- --target=arm64`

wdyt?

Reply:
I thought about that, wouldn't be a bit confusing considering that is something valid only for Android (so `target_os=android`) ? Also, `--output-directory` would be the main command, but there are a few other useful options like -v for verbose mode, or a specific sdk path with --sdk-path. These could be handy for debugging purposes.

Reply:
![image](https://github.com/brave/brave-core/assets/3308503/9a69d264-0427-4029-83fe-0986c7b3275b)

Reply:
> I thought about that, wouldn't be a bit confusing considering that is something valid only for Android (so `target_os=android`) ? Also, `--output-directory` would be the main command, but there are a few other useful options like -v for verbose mode, or a specific sdk path with --sdk-path. These could be handy for debugging purposes.

sure, but we can pass other parameters as is. We need to only "generate" `--output-directory` one using our known approach. Thankfully `--target_arch` is not used by `generate_gradle.py`.

Reply:
I've implemented a possible solution in the commit [`ad22da1` (#19385)](https://github.com/brave/brave-core/pull/19385/commits/ad22da1f25d5f8acb6d33be4009eaa834a69d05f).

This approach feels a bit convoluted, but maybe it's just me that was used to type `--output-directory` easily. This approach also required some trickery because the `allowUnknownOption` was also including the `--target_arch` parameter and the Gradle script was failing, so I ended up filtering the  `passthroughArgs`.

Let me know what you think!

Reply:
> I've implemented a possible solution in the commit [`ad22da1` (#19385)](https://github.com/brave/brave-core/pull/19385/commits/ad22da1f25d5f8acb6d33be4009eaa834a69d05f).
> 
> This approach feels a bit convoluted, but maybe it's just me that was used to type `--output-directory` easily. This approach also required some trickery because the `allowUnknownOption` was also including the `--target_arch` parameter and the Gradle script was failing, so I ended up filtering the `passthroughArgs`.
> 
> Let me know what you think!

logged https://github.com/brave/brave-browser/issues/31944